### DISCUSSION TITLE
Future of rescript support

### DISCUSSION THREAD

**Discussion by DewarM** [parcel-bundler/parcel#5162] [General]
Hi folks 👋 

I have been starting out with both Parcel and rescript (https://rescript-lang.org). I have seen that ReasonML (rescript's precursor prior to its rebrand?). I see that the following file formats are supported: `ml` & `re`. Initially I had thought that this could easily expanded to include the new `res`, on reading: https://rescript-lang.org/docs/manual/latest/interop-with-js-build-systems#popular-js-build-systems I am no longer sure if this is a good idea as the rescript authors actively discourage this kind of behaviour. 

In any case, I've opened a draft pulled request here where I have tried to make the required changes for `res` files to function. As I said previously, just starting with parcel so not 100% sure that I have covered everything. 

PR: https://github.com/parcel-bundler/parcel/pull/5161

**Comment by lukad** [comment]
I would love to get this merged. In my opinion loaders should not be discouraged as they are user friendly and let you get actual work done instead of having to fiddle with your build pipeline, which is probably more error prone.

**Comment by cdaringe** [comment]
The rescript/reason compiler are the same compiler, and is backwards compatible. Let's merge it!
